### ISSUE TITLE
need more helpful error output for git dependencies upon install failure

### ISSUE THREAD

**Issue by jonblatho** [oven-sh/bun#2342]
### What platform is your computer?

Darwin 22.3.0 arm64 arm

### How did you attempt to install or upgrade?

`bun install` and `bun install --verbose`

### What do you see instead?

```
error: "git clone" for "[git dependency package name]" failed
error: "git clone" for "[another git dependency package name]" failed
```

This error output is unfortunately not at all helpful in understanding what went wrong, whose fault it is, or how I might go about resolving the error.

### Additional information

 Example structure of impacted packages in package.json (which installs fine using either `npm i` or `yarn`):
```
{
...
  "dependencies": {
    "private-package": "git+ssh://git@our-gitlab-hostname.com:path/to/repo.git#x.y.z"
  }
...
}
```

These repositories are not publicly accessible; either HTTPS or SSH authentication with valid credentials is required to access the impacted packages. Many of these packages themselves have private git dependencies structured the same as in the example above. I am able to manually run `git clone git@our-gitlab-hostname.com:path/to/repo.git` for these repositories without any trouble.

I did also try it with a random public GitHub repository for a NPM package and encountered the same failure. However, _only_ git dependencies fail to install. The failure occurs very quickly.

**Comment by Electroid** [comment]
Agreed, we should make these errors more specific. Thanks for the report

**Comment by Electroid** [comment]
@alexlamsl I believe you made this change?

**Comment by thallada** [comment]
I'm getting the same issue trying to `bun install` with a `package.json` that includes a dependency on a private github repo. e.g. (`git+ssh://git@github.com:foo/bar.git`). `npm install` has no trouble with this. Is there some config I'm missing or does bun currently not support installing private `git+ssh` dependencies?

**Comment by Jarred-Sumner** [comment]
@alexlamsl does it go through the github api for git+ssh://github.com? maybe we should force it to go through git cli
